ddtrace/profiling/exporter/pprof.pyx

```diff
@@ -190,9 +191,12 @@ class _PprofConverter(object):
 
     def _to_Function(
         self,
-        filename: str,
-        funcname: str,
-    ) -> pprof_FunctionType:
+        filename,  # type: str
+        funcname,  # type: str
+    ):
+        # type: (...) -> pprof_FunctionType
+        filename = sanitize_string(filename)
+        funcname = sanitize_string(funcname)
```

Comment:
The typing here is a bit funny now. I suppose we no longer expect the arguments to be `str`, otherwise `sanitize_string` would be redundant. So perhaps we should do
```suggestion
        filename,  # type: t.Union[str, bytes]
        funcname,  # type: t.Union[str, bytes]
    ):
        # type: (...) -> pprof_FunctionType
        _filename = sanitize_string(filename)
        _funcname = sanitize_string(funcname)
```
?

Reply:
I modified the code so as to adhere to the requirement that `str` is the input type.  This is a convention that has held true for a long time and I don't _think_ there's any prevailing reason why filename/funcname should suddenly be _bytes_ in particular.  I would suspect we're seeing the same behavior we've seen before, which is that frame/code members (or objects) are becoming random types.

What do you think?  Should we move to explicitly support bytes (in which case I'll update `sanitize_str()` to perform this conversion again), or should we be more particular?

Reply:
Ah so we are opening up to the possibility of effectively receiving `Any`. I would then change the type of the arguments to `Any` (even though we'd really hope for them to be `str`).

Reply:
Yep, I see your point. Let's keep the current typing, but let's also make it clear in the implementation that we are working around the runtime 🙏

Reply:
Sounds good